src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests.cs

```diff
@@ -1029,6 +1029,39 @@ void Test12()
             VerifyNotInScope(model, x12Ref[1]);
         }
 
+        [Fact]
+        public void PropertyNamedInComplexPattern()
+        {
+            var source =
+@"
+using System;
+public class Program
+{
+    public static void Main()
+    {
+        object o = nameof(Main);
+        Console.WriteLine(o is string { Length is 4 });
+    }
+}
+";
+            var compilation = CreateCompilationWithMscorlib45(source, options: TestOptions.DebugExe, parseOptions: patternParseOptions);
+
+            compilation.VerifyDiagnostics();
+            var tree = compilation.SyntaxTrees.Single();
+            var model = compilation.GetSemanticModel(tree);
+            var propPats = tree.GetRoot().DescendantNodes().OfType<SubPropertyPatternSyntax>().ToArray();
+            foreach (var p in propPats)
+            {
+                var si = model.GetSymbolInfo(p);
+                Assert.NotNull(si.Symbol);
+                Assert.Equal("Length", si.Symbol.Name);
+                si = model.GetSymbolInfo(p.Left);
+                Assert.NotNull(si.Symbol);
+                Assert.Equal("Length", si.Symbol.Name);
+            }
+
+        }
```

Comment:
All properties of SimbolInfo should be verified.
There should be tests for SubPropertyPatternSyntax.Left passed to GetSymbol info.
There should be tests for all these nodes passed as an ExpressionSyntax.
There should be tests for error scenarios.